test/ssl_test.c

```diff
@@ -196,20 +196,17 @@ static int check_nid(const char *name, int expected_nid, int nid)
 
 static void print_ca_names(STACK_OF(X509_NAME) *names)
 {
-    BIO *err;
     int i;
 
     if (names == NULL || sk_X509_NAME_num(names) == 0) {
-        fprintf(stderr, "    <empty>\n");
+        TEST_note("    <empty>");
         return;
     }
-    err = BIO_new_fp(stderr, BIO_NOCLOSE);
     for (i = 0; i < sk_X509_NAME_num(names); i++) {
-        X509_NAME_print_ex(err, sk_X509_NAME_value(names, i), 4,
+        X509_NAME_print_ex(bio_err, sk_X509_NAME_value(names, i), 4,
                            XN_FLAG_ONELINE);
-        BIO_puts(err, "\n");
+        BIO_puts(bio_err, "\n");
     }
-    BIO_free(err);
```

Comment:
I scratched out a BIO filter in some free time today.  It isn't working properly yet.

Should it be counting spaces before the first `#` on each line as the indent (if present & remembering the previous if not) or should it indent as per the current indent level?  Both are incorrect of course, it is more a question of which is better.

I'm also thinking of _either_ only indenting stderr _or_ allowing the test framework a capability to bypass the filter.  This is so that the "ok" or "not ok" messages will not be hashed out.